### PR TITLE
added notification for GA

### PR DESCRIPTION
**Pull request by Rohithone**
# PR Checklist

Please check if your PR fulfills the following requirements
- [ ] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

# What is the ticket # detailing the issue?

Ticket: https://redmine.priv.dchbx.org/issues/101340

# A brief description of the changes

Current behavior: message notification only sending to broker agency, not to the employer and general agency

New behavior: updated code to send a message notification to employer and general agency along with broker agency

# Environment Variable

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. Please share the name of the variable below that would enable/disable the feature and which client it applies to.

Variable name:

- [ ] DC
- [ ] ME

### CODE REVIEW

**Review by harshaellanki — DISMISSED** [review]

File: components/benefit_sponsors/app/models/benefit_sponsors/services/invitation_email_service.rb [hunk 1]
```diff
@@ -25,6 +26,15 @@ def send_broker_successfully_associated_email
         create_secure_message(message_params, broker_person, :inbox)
       end
 
+      def send_general_agency_successfully_associated_email
+        broker_agency_profile = BenefitSponsors::Organizations::BrokerAgencyProfile.find(get_bson_id(broker_agency_profile_id))
+        general_agency = broker_agency_profile.default_general_agency_profile
+        subject = "You are associated to #{broker_agency_profile.organization.legal_name}- #{general_agency.legal_name} (Hire)"
```
Comment (polographer) [inline components/benefit_sponsors/app/models/benefit_sponsors/services/invitation_email_service.rb:32]: this is a hardcoded string, can you re-write it to use the translation files, next line has the same issue

File: components/benefit_sponsors/spec/models/benefit_sponsors/services/broker_management_service_spec.rb [hunk 2]
```diff
@@ -38,13 +45,30 @@ module BenefitSponsors
...
         subject.assign_agencies(broker_management_form_create)
       end
 
       it 'should return true once it succesfully assigns broker agency to the employer_profile' do
         expect(subject.assign_agencies(broker_management_form_create)).to be_truthy
       end
 
+      it 'should send a message to the general_agency' do
+        general_agency_profile.reload
+        subject = "You are associated to #{broker_agency_profile1.organization.legal_name}- #{general_agency_profile.legal_name} (Hire)"
```
Comment (polographer) [inline components/benefit_sponsors/spec/models/benefit_sponsors/services/broker_management_service_spec.rb:58]: same hardcoded string issue

**Review by polographer — CHANGES_REQUESTED** [review]

**Comment by polographer** [comment]
can you also re-apply the PR description template of the repo

**Comment by Rohithone** [comment]
All issues are corrected.

File: components/benefit_sponsors/spec/models/benefit_sponsors/services/broker_management_service_spec.rb (hunk 2 above)
Comment (Rohithone) [reply components/benefit_sponsors/spec/models/benefit_sponsors/services/broker_management_service_spec.rb]: Fixed.

File: components/benefit_sponsors/app/models/benefit_sponsors/services/broker_management_service.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def assign_agencies_for_employer(form)
         @employer_profile = BenefitSponsors::Organizations::Profile.find(form.employer_profile_id)
         assign_broker_agency_for_emp(form.broker_agency_profile_id, form.broker_role_id)
         # # TODO fix this during notices implementation
```
[the remark this answers is not in the document]
Comment (Rohithone) [reply components/benefit_sponsors/app/models/benefit_sponsors/services/broker_management_service.rb]: Done.

File: components/benefit_sponsors/app/models/benefit_sponsors/services/invitation_email_service.rb (hunk 1 above)
Comment (Rohithone) [reply components/benefit_sponsors/app/models/benefit_sponsors/services/invitation_email_service.rb]: Changes added.

**Review by harshaellanki — APPROVED** [review]

**Review by polographer — APPROVED** [review]

**Review by polographer — APPROVED** [review]

**Pull request merged**